mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -445,6 +513,7 @@ struct TransposeRewritePattern : public OpRewritePattern<tosa::TransposeOp> {
                                              Value tOutput,
                                              ArrayRef<int32_t> dims,
                                              Value tInput) const {
+    SmallVector<Operation *> toBeErased;
```

Comment:
While debugging, I wandered down the address-sanitiser path.  Didn't find my problem that way, but I did see that nodes erased here were sometimes used after erasure.  Collect them and then erase at the end.

Reply:
Yeah no, let's actually debug that - but maybe in a future ticket